packages/nakama-js/client.ts

```diff
@@ -611,6 +611,13 @@ export class Client {
 
     fetchOptions.headers = {...headers};
 
+    const descriptor = Object.getOwnPropertyDescriptor(XMLHttpRequest.prototype, "withCredentials");
```

Comment:
All of the `Authenticate...` functions in `Client.ts` bypass the generated core, which is why we have all these duplicate descriptor checks.

The body of these auth functions is very similar to what we're doing in the generated core, so can give that a careful update if we want to avoid this.

Reply:
Nice one @lugehorsam lets definitely give the careful update a go for these functions. I think they were originally separated out because of limitations in the code generator tool.